lib/zendesk_apps_tools/deploy.rb

```diff
@@ -112,7 +118,18 @@ def check_job(job_id)
         sleep 3
       end
     rescue Faraday::Error::ClientError => e
-      say_error_and_exit e.message
+      delete_cache_exit_and_error(e.message)
+    end
+
+    private
+
+    def delete_cache_exit_and_error(error_message)
+      cache.delete(['app_id', 'subdomain', 'username'])
```

Comment:
Should we output a message saying we're deleting the user cache because of the failure?

Reply:
@danielbreves seems a bit overkill. Why would you want it to store potentially incorrect user inputs? As a developer/user of this tool I'm not sure I'd care the cache is being deleted but rather than they're just cached after a successful connection has been established once